lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -1023,6 +1022,39 @@ namespace {
         return eed->getAttrs().hasAttribute<DowngradeExhaustivityCheckAttr>();
       }
 
+      static void
```

Comment:
I could use a comment explaining where the output is and why this is specialized from decompose, if that's what's going on.

Reply:
Also a high-level explanation of the algorithm would help me.